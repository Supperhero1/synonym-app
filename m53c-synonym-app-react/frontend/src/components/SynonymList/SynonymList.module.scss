.synonymListWrapper {
  h2 {
    text-align: center;
    .selectedWord {
      color: deepskyblue;
    }
  }
  .synonymList {
    list-style: none;
    flex-wrap: wrap;
    display: flex;
    flex-direction: row;
    justify-content: center;
    padding: 0;
    font-size: 1.3rem;
    max-width: 90vw;
  }
}
