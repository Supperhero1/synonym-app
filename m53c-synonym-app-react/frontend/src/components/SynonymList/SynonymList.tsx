import styles from './SynonymList.module.scss'
import Tumbleweeds from "./Tumbleweeds/Tumbleweeds";
import Synonym from "./Synonym/Synonym";

export default ({
    originalWord,
    synonyms,
    deleteHandler
}: {
    originalWord: string,
    synonyms: string[],
    deleteHandler: (word: string) => Promise<void>
}) => {
    return <section className={`${styles.synonymListWrapper} ${synonyms.length ? '' : styles.hidden}`}>
        <h2>{`Displaying synonyms for the word `}<span className={styles.selectedWord} >{`${originalWord.toLocaleUpperCase()}`}</span>:</h2>
        <ul className={styles.synonymList}>
            {
                synonyms.length
                    ? synonyms.map((synonym) => <Synonym word={synonym} deleteHandler={deleteHandler} />)
                    : <Tumbleweeds />
            }
        </ul>
    </section>
}
