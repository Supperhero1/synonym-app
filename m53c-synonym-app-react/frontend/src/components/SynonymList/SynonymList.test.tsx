import { render } from "@testing-library/react";
import React from "react";
import SynonymList from "./SynonymList";

it('renders correctly', () => {
    const tree = render(
        <SynonymList
            originalWord={'word'}
            synonyms={[ 'one', 'two', 'three' ]}
            deleteHandler={(() => { return Promise.resolve() })}
        />)

    expect(tree).toMatchSnapshot()
})
