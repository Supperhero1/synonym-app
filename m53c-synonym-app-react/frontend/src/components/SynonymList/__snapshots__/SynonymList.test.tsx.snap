// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <section
        class="synonymListWrapper "
      >
        <h2>
          Displaying synonyms for the word 
          <span
            class="selectedWord"
          >
            WORD
          </span>
          :
        </h2>
        <ul
          class="synonymList"
        >
          <li
            class="synonymListItem"
          >
            <div
              class="deleteButton"
            >
              <svg
                aria-hidden="true"
                class="svg-inline--fa fa-trash deleteIcon"
                data-icon="trash"
                data-prefix="fas"
                focusable="false"
                role="img"
                viewBox="0 0 448 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M135.2 17.7L128 32H32C14.3 32 0 46.3 0 64S14.3 96 32 96H416c17.7 0 32-14.3 32-32s-14.3-32-32-32H320l-7.2-14.3C307.4 6.8 296.3 0 284.2 0H163.8c-12.1 0-23.2 6.8-28.6 17.7zM416 128H32L53.2 467c1.6 25.3 22.6 45 47.9 45H346.9c25.3 0 46.3-19.7 47.9-45L416 128z"
                  fill="currentColor"
                />
              </svg>
            </div>
            one
          </li>
          <li
            class="synonymListItem"
          >
            <div
              class="deleteButton"
            >
              <svg
                aria-hidden="true"
                class="svg-inline--fa fa-trash deleteIcon"
                data-icon="trash"
                data-prefix="fas"
                focusable="false"
                role="img"
                viewBox="0 0 448 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M135.2 17.7L128 32H32C14.3 32 0 46.3 0 64S14.3 96 32 96H416c17.7 0 32-14.3 32-32s-14.3-32-32-32H320l-7.2-14.3C307.4 6.8 296.3 0 284.2 0H163.8c-12.1 0-23.2 6.8-28.6 17.7zM416 128H32L53.2 467c1.6 25.3 22.6 45 47.9 45H346.9c25.3 0 46.3-19.7 47.9-45L416 128z"
                  fill="currentColor"
                />
              </svg>
            </div>
            two
          </li>
          <li
            class="synonymListItem"
          >
            <div
              class="deleteButton"
            >
              <svg
                aria-hidden="true"
                class="svg-inline--fa fa-trash deleteIcon"
                data-icon="trash"
                data-prefix="fas"
                focusable="false"
                role="img"
                viewBox="0 0 448 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M135.2 17.7L128 32H32C14.3 32 0 46.3 0 64S14.3 96 32 96H416c17.7 0 32-14.3 32-32s-14.3-32-32-32H320l-7.2-14.3C307.4 6.8 296.3 0 284.2 0H163.8c-12.1 0-23.2 6.8-28.6 17.7zM416 128H32L53.2 467c1.6 25.3 22.6 45 47.9 45H346.9c25.3 0 46.3-19.7 47.9-45L416 128z"
                  fill="currentColor"
                />
              </svg>
            </div>
            three
          </li>
        </ul>
      </section>
    </div>
  </body>,
  "container": <div>
    <section
      class="synonymListWrapper "
    >
      <h2>
        Displaying synonyms for the word 
        <span
          class="selectedWord"
        >
          WORD
        </span>
        :
      </h2>
      <ul
        class="synonymList"
      >
        <li
          class="synonymListItem"
        >
          <div
            class="deleteButton"
          >
            <svg
              aria-hidden="true"
              class="svg-inline--fa fa-trash deleteIcon"
              data-icon="trash"
              data-prefix="fas"
              focusable="false"
              role="img"
              viewBox="0 0 448 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M135.2 17.7L128 32H32C14.3 32 0 46.3 0 64S14.3 96 32 96H416c17.7 0 32-14.3 32-32s-14.3-32-32-32H320l-7.2-14.3C307.4 6.8 296.3 0 284.2 0H163.8c-12.1 0-23.2 6.8-28.6 17.7zM416 128H32L53.2 467c1.6 25.3 22.6 45 47.9 45H346.9c25.3 0 46.3-19.7 47.9-45L416 128z"
                fill="currentColor"
              />
            </svg>
          </div>
          one
        </li>
        <li
          class="synonymListItem"
        >
          <div
            class="deleteButton"
          >
            <svg
              aria-hidden="true"
              class="svg-inline--fa fa-trash deleteIcon"
              data-icon="trash"
              data-prefix="fas"
              focusable="false"
              role="img"
              viewBox="0 0 448 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M135.2 17.7L128 32H32C14.3 32 0 46.3 0 64S14.3 96 32 96H416c17.7 0 32-14.3 32-32s-14.3-32-32-32H320l-7.2-14.3C307.4 6.8 296.3 0 284.2 0H163.8c-12.1 0-23.2 6.8-28.6 17.7zM416 128H32L53.2 467c1.6 25.3 22.6 45 47.9 45H346.9c25.3 0 46.3-19.7 47.9-45L416 128z"
                fill="currentColor"
              />
            </svg>
          </div>
          two
        </li>
        <li
          class="synonymListItem"
        >
          <div
            class="deleteButton"
          >
            <svg
              aria-hidden="true"
              class="svg-inline--fa fa-trash deleteIcon"
              data-icon="trash"
              data-prefix="fas"
              focusable="false"
              role="img"
              viewBox="0 0 448 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M135.2 17.7L128 32H32C14.3 32 0 46.3 0 64S14.3 96 32 96H416c17.7 0 32-14.3 32-32s-14.3-32-32-32H320l-7.2-14.3C307.4 6.8 296.3 0 284.2 0H163.8c-12.1 0-23.2 6.8-28.6 17.7zM416 128H32L53.2 467c1.6 25.3 22.6 45 47.9 45H346.9c25.3 0 46.3-19.7 47.9-45L416 128z"
                fill="currentColor"
              />
            </svg>
          </div>
          three
        </li>
      </ul>
    </section>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
